### ISSUE TITLE
[Swift] ./bin/swift-petstore.sh fails on master

### ISSUE THREAD

**Issue by jgavris** [swagger-api/swagger-codegen#3265]
##### Description

Swift Petstore example code generation fails on `master` (

##### Swagger-codegen version

`45f13b2`

##### Swagger declaration file content or url

`./modules/swagger-codegen/src/test/resources/2_0/petstore.json`

##### Command line used for generation

`git checkout 45f13b2 && mvn clean package && ./bin/swift-petstore.sh`

```shell
Exception in thread "main" java.lang.RuntimeException: Could not process operation:
  Tag: pet
  Operation: findPetsByStatus
  Resource: get /pet/findByStatus
  Definitions: {User=io.swagger.models.ModelImpl@41b446b7, Category=io.swagger.models.ModelImpl@b0b183eb, Pet=io.swagger.models.ModelImpl@989b6400, Tag=io.swagger.models.ModelImpl@94631c47, Order=io.swagger.models.ModelImpl@17904dd5}
  Exception: null
	at io.swagger.codegen.DefaultGenerator.processOperation(DefaultGenerator.java:786)
	at io.swagger.codegen.DefaultGenerator.processPaths(DefaultGenerator.java:678)
	at io.swagger.codegen.DefaultGenerator.generate(DefaultGenerator.java:366)
	at io.swagger.codegen.cmd.Generate.run(Generate.java:223)
	at io.swagger.codegen.SwaggerCodegen.main(SwaggerCodegen.java:36)
Caused by: java.lang.NullPointerException
	at io.swagger.codegen.languages.SwiftCodegen.fromProperty(SwiftCodegen.java:330)
	at io.swagger.codegen.DefaultCodegen.fromParameter(DefaultCodegen.java:2140)
	at io.swagger.codegen.DefaultCodegen.fromOperation(DefaultCodegen.java:1906)
	at io.swagger.codegen.languages.SwiftCodegen.fromOperation(SwiftCodegen.java:447)
	at io.swagger.codegen.DefaultGenerator.processOperation(DefaultGenerator.java:734)
	... 4 more
```

##### Steps to reproduce

`git checkout 45f13b2 && mvn clean package && ./bin/swift-petstore.sh`

##### Suggest a Fix

I'm a bit new to Swagger. Getting the integration tests fixed and also running again for Objective-C and Swift would be rad. If it's difficult to test both JDK 7/8 and OS X in the same `.travis.yml`, maybe an easy solution would be to add another CI service (CircleCI) and configure that one for OS X profiles.

**Comment by wing328** [comment]
@jgavris thanks for reporting the issue. I'm looking into it.

**Comment by wing328** [comment]
@jgavris it's due to enum enhancement I did earlier which breaks how swift handles enum. I'm working on a fix right now.

**Comment by wing328** [comment]
@jgavris I've filed #3267 to fix the NPE and will merge into master after CI tests pass.

**Comment by wing328** [comment]
@jgavris PR merged into master. Please pull the latest to give it a try.

**Comment by jgavris** [comment]
@wing328 Thanks! Possible to get the swift / objc tests uncommented out of the CI test modules?

```xml
                <!--module>samples/client/petstore/objc/SwaggerClientTests</module-->
                <!--module>samples/client/petstore/swift/SwaggerClientTests</module-->
```

**Comment by wing328** [comment]
@jgavris I wanted to but Travis CI does not support multiple `langauges` at the moment: https://docs.travis-ci.com/user/languages/objective-c. A possible workaround is to manually install `xctool` but I didn't have time to look into that yet.

**Comment by jgavris** [comment]
How about using CircleCI as a second CI service, for the Xcode toolchain?

**Comment by wing328** [comment]
Good idea. I've only played with shippable as a secondary CI service to cover ObjC and Swift API client but didn't have much progress so far.

Did you have experience with CircleCI? Do you have cycle to contribute a `circle.yml` to vero both ObjC and Swift Petstore API client?

**Comment by jgavris** [comment]
@wing328 Not specifically CircleCI, but I can figure it out. Would you mind doing the repository / organization setup for this repo on CircleCI? 

https://github.com/integrations/circle-ci

I will push a PR with a `circle.yml` when that's ready to test.

**Comment by wing328** [comment]
I need to check as I may not have the right. 

I think you can test with the folk repo of swagger-codegen (https://github.com/jgavris/swagger-codegen.git) to begin with.

**Comment by wing328** [comment]
@jgavris  I've added "appveyor" to cover testing in the Windows environment via https://github.com/swagger-api/swagger-codegen/pull/3275.

Confirmed that I do not have the right to add any services to this project.

Once the CI tests for Objc/Swift is ready with CircleCI or other CI providers, we can ask Tony Tam to help set it up in the Github repo setting page.

**Comment by jgavris** [comment]
@wing328 Unfortunately CircleCI charges for OS X agents. Maybe we could convert the travis language to OS X and run the JDK 7 / 8 tests in docker images on OS X? Continuing on with your efforts here https://github.com/swagger-api/swagger-codegen/pull/3411

**Comment by wing328** [comment]
@jgavris thanks for the update. Let me give that a try later this week.
